From 612506947caa27bdc1d7cdc30f9fe1fb8e6f8ee5 Mon Sep 17 00:00:00 2001
From: "Radu Pirea (NXP OSS)" <radu-nicolae.pirea@oss.nxp.com>
Date: Wed, 9 Feb 2022 09:59:49 +0200
Subject: [PATCH 03/51] pfeng: remove board defines

Remove board defines. The user should be able to configure any interface
in rgmii mode. These kind of configurations should not be restricted at
build time.

Issue: ALB-8367

Upstream-Status: Pending 

Signed-off-by: Radu Pirea (NXP OSS) <radu-nicolae.pirea@oss.nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 configs/s32g274ardb2_defconfig      |  1 -
 configs/s32g274ardb2_qspi_defconfig |  1 -
 configs/s32g2xxaevb_defconfig       |  1 -
 configs/s32g2xxaevb_qspi_defconfig  |  1 -
 configs/s32g399ardb3_defconfig      |  1 -
 configs/s32g399ardb3_qspi_defconfig |  1 -
 configs/s32g3xxaevb_defconfig       |  1 -
 configs/s32g3xxaevb_qspi_defconfig  |  1 -
 drivers/net/Kconfig                 | 18 ---------------
 drivers/net/pfeng/pfeng_cmd.c       | 36 -----------------------------
 10 files changed, 62 deletions(-)

diff --git a/configs/s32g274ardb2_defconfig b/configs/s32g274ardb2_defconfig
index 4f0cae576f..96c29a0589 100644
--- a/configs/s32g274ardb2_defconfig
+++ b/configs/s32g274ardb2_defconfig
@@ -30,7 +30,6 @@ CONFIG_NET_RANDOM_ETHADDR=y
 # CONFIG_DM_DEVICE_REMOVE is not set
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g274ardb2_qspi_defconfig b/configs/s32g274ardb2_qspi_defconfig
index e3735fa9cc..f65b867b77 100644
--- a/configs/s32g274ardb2_qspi_defconfig
+++ b/configs/s32g274ardb2_qspi_defconfig
@@ -35,7 +35,6 @@ CONFIG_NET_RANDOM_ETHADDR=y
 # CONFIG_DM_DEVICE_REMOVE is not set
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g2xxaevb_defconfig b/configs/s32g2xxaevb_defconfig
index 24c643e4d8..a83d4f6ada 100644
--- a/configs/s32g2xxaevb_defconfig
+++ b/configs/s32g2xxaevb_defconfig
@@ -33,7 +33,6 @@ CONFIG_SYS_EEPROM_SIZE=128
 CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=10
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g2xxaevb_qspi_defconfig b/configs/s32g2xxaevb_qspi_defconfig
index f134146404..93205557ab 100644
--- a/configs/s32g2xxaevb_qspi_defconfig
+++ b/configs/s32g2xxaevb_qspi_defconfig
@@ -38,7 +38,6 @@ CONFIG_SYS_EEPROM_SIZE=128
 CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=10
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g399ardb3_defconfig b/configs/s32g399ardb3_defconfig
index 41edb09cd0..c298949c6a 100644
--- a/configs/s32g399ardb3_defconfig
+++ b/configs/s32g399ardb3_defconfig
@@ -31,7 +31,6 @@ CONFIG_NET_RANDOM_ETHADDR=y
 # CONFIG_DM_DEVICE_REMOVE is not set
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g399ardb3_qspi_defconfig b/configs/s32g399ardb3_qspi_defconfig
index b439a67eee..514b035c18 100644
--- a/configs/s32g399ardb3_qspi_defconfig
+++ b/configs/s32g399ardb3_qspi_defconfig
@@ -37,7 +37,6 @@ CONFIG_NET_RANDOM_ETHADDR=y
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
 CONFIG_FSL_PFENG_FW_LOC_SDCARD=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g3xxaevb_defconfig b/configs/s32g3xxaevb_defconfig
index c18d643d3b..3a761f3a99 100644
--- a/configs/s32g3xxaevb_defconfig
+++ b/configs/s32g3xxaevb_defconfig
@@ -33,7 +33,6 @@ CONFIG_SYS_EEPROM_SIZE=128
 CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=10
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/configs/s32g3xxaevb_qspi_defconfig b/configs/s32g3xxaevb_qspi_defconfig
index 681babc068..0cde06d692 100644
--- a/configs/s32g3xxaevb_qspi_defconfig
+++ b/configs/s32g3xxaevb_qspi_defconfig
@@ -39,7 +39,6 @@ CONFIG_SYS_EEPROM_SIZE=128
 CONFIG_SYS_EEPROM_PAGE_WRITE_DELAY_MS=10
 # CONFIG_SPI_FLASH_BAR is not set
 CONFIG_PHY_FIXED=y
-CONFIG_FSL_PFENG_EMAC_1_RGMII=y
 CONFIG_E1000=y
 CONFIG_CMD_E1000=y
 CONFIG_RGMII=y
diff --git a/drivers/net/Kconfig b/drivers/net/Kconfig
index e5455deb86..8d42d15d22 100644
--- a/drivers/net/Kconfig
+++ b/drivers/net/Kconfig
@@ -118,24 +118,6 @@ config FSL_PFENG_FW_PART
 	  For "sdcard", it is usually the first device's first (boot) partition, so "0:1".
 	  For "qspi" it is the addrees of reserved memory partition.
 
-config FSL_PFENG_EMAC_0_RGMII
-	bool "Allow RGMII on PFE EMAC_0"
-	depends on TARGET_S32G2XXAEVB || NXP_S32GRDB_BOARD || TARGET_S32G3XXAEVB
-	default n
-	help
-	  PFE EMAC_0 and EMAC_2 share RGMII interface on S32G-VNP-PROC
-	  and S32G-VNP-RDB boards. Enable possibility to switch EMAC_0
-	  to RGMII.
-
-config FSL_PFENG_EMAC_1_RGMII
-	bool "Allow RGMII on PFE EMAC 1"
-	depends on TARGET_S32G2XXAEVB || NXP_S32GRDB_BOARD || TARGET_S32G3XXAEVB
-	default n
-	help
-	  PFE EMAC_1 and GMAC share RGMII interface on S32G-VNP-PROC
-	  and S32G-VNP-RDB boards. Enable possibility to switch EMAC_1
-	  to RGMII.
-
 endif # FSL_PFENG
 
 config PHY_GIGE
diff --git a/drivers/net/pfeng/pfeng_cmd.c b/drivers/net/pfeng/pfeng_cmd.c
index 3ad6ee1602..200e17d890 100644
--- a/drivers/net/pfeng/pfeng_cmd.c
+++ b/drivers/net/pfeng/pfeng_cmd.c
@@ -251,14 +251,8 @@ static void set_pfe_mac0_clk(int intf0, struct udevice *pfe_dev)
 		tx = "mac0_tx_sgmii";
 		break;
 	case PHY_INTERFACE_MODE_RGMII:
-#if CONFIG_IS_ENABLED(TARGET_S32G2XXAEVB) || \
-		CONFIG_IS_ENABLED(NXP_S32GRDB_BOARD) || \
-		CONFIG_IS_ENABLED(TARGET_S32G3XXAEVB)
-#if CONFIG_IS_ENABLED(FSL_PFENG_EMAC_0_RGMII)
 		rx = "mac0_rx_rgmii";
 		tx = "mac0_tx_rgmii";
-#endif
-#endif
 		break;
 	case PHY_INTERFACE_MODE_RMII:
 	case PHY_INTERFACE_MODE_MII:
@@ -293,14 +287,8 @@ static void set_pfe_mac1_clk(int intf1, struct udevice *pfe_dev)
 		tx = "mac1_tx_sgmii";
 		break;
 	case PHY_INTERFACE_MODE_RGMII:
-#if CONFIG_IS_ENABLED(TARGET_S32G2XXAEVB) || \
-		CONFIG_IS_ENABLED(NXP_S32GRDB_BOARD) || \
-		CONFIG_IS_ENABLED(TARGET_S32G3XXAEVB)
-#if CONFIG_IS_ENABLED(FSL_PFENG_EMAC_1_RGMII)
 		rx = "mac1_rx_rgmii";
 		tx = "mac1_tx_rgmii";
-#endif
-#endif
 		break;
 
 	case PHY_INTERFACE_MODE_RMII:
@@ -337,14 +325,8 @@ static void set_pfe_mac2_clk(int intf2, struct udevice *pfe_dev)
 		tx = "mac2_tx_sgmii";
 		break;
 	case PHY_INTERFACE_MODE_RGMII:
-#if CONFIG_IS_ENABLED(TARGET_S32G2XXAEVB) || \
-		CONFIG_IS_ENABLED(NXP_S32GRDB_BOARD) || \
-		CONFIG_IS_ENABLED(TARGET_S32G3XXAEVB)
-#if !CONFIG_IS_ENABLED(FSL_PFENG_EMAC_0_RGMII)
 		rx = "mac2_rx_rgmii";
 		tx = "mac2_tx_rgmii";
-#endif
-#endif
 		break;
 	case PHY_INTERFACE_MODE_RMII:
 	case PHY_INTERFACE_MODE_MII:
@@ -425,13 +407,7 @@ static void setup_iomux_pfe(struct udevice *dev,
 		break;
 
 	case PHY_INTERFACE_MODE_RGMII:
-#if CONFIG_IS_ENABLED(TARGET_S32G2XXAEVB) || \
-		CONFIG_IS_ENABLED(NXP_S32GRDB_BOARD) || \
-		CONFIG_IS_ENABLED(TARGET_S32G3XXAEVB)
-#if CONFIG_IS_ENABLED(FSL_PFENG_EMAC_0_RGMII)
 		pinctrl_select_state(dev, "pfe0_rgmii");
-#endif
-#endif
 		break;
 
 	case PHY_INTERFACE_MODE_RMII:
@@ -454,13 +430,7 @@ static void setup_iomux_pfe(struct udevice *dev,
 		break;
 
 	case PHY_INTERFACE_MODE_RGMII:
-#if CONFIG_IS_ENABLED(TARGET_S32G2XXAEVB) || \
-		CONFIG_IS_ENABLED(NXP_S32GRDB_BOARD) || \
-		CONFIG_IS_ENABLED(TARGET_S32G3XXAEVB)
-#if CONFIG_IS_ENABLED(FSL_PFENG_EMAC_1_RGMII)
 		pinctrl_select_state(dev, "pfe1_rgmii");
-#endif
-#endif
 		break;
 
 	case PHY_INTERFACE_MODE_RMII:
@@ -483,13 +453,7 @@ static void setup_iomux_pfe(struct udevice *dev,
 		break;
 
 	case PHY_INTERFACE_MODE_RGMII:
-#if CONFIG_IS_ENABLED(TARGET_S32G2XXAEVB) || \
-		CONFIG_IS_ENABLED(NXP_S32GRDB_BOARD) || \
-		CONFIG_IS_ENABLED(TARGET_S32G3XXAEVB)
-#if !CONFIG_IS_ENABLED(FSL_PFENG_EMAC_0_RGMII)
 		pinctrl_select_state(dev, "pfe2_rgmii");
-#endif
-#endif
 		break;
 
 	case PHY_INTERFACE_MODE_RMII:
-- 
2.17.1

